pkg/controllers/managementuser/rbac/prtb_handler.go

```diff
@@ -271,20 +271,31 @@ func (m *manager) checkForGlobalResourceRules(role *v3.RoleTemplate, resource st
 }
 
 // Ensure the clusterRole used to grant access of global resources to users/groups in projects has appropriate rules for the given resource and verbs
-func (m *manager) reconcileRoleForProjectAccessToGlobalResource(resource string, rt *v3.RoleTemplate, newVerbs map[string]bool) (string, error) {
+func (m *manager) reconcileRoleForProjectAccessToGlobalResource(resource string, rt *v3.RoleTemplate, newVerbs map[string]bool, baseRule rbacv1.PolicyRule) (string, error) {
 	clusterRoles := m.workload.RBAC.ClusterRoles("")
 	roleName := rt.Name + "-promoted"
 	if role, err := m.crLister.Get("", roleName); err == nil && role != nil {
 		currentVerbs := map[string]bool{}
+		currentResourceNames := map[string]struct{}{}
```

Comment:
Asked Ryan and there is no point in the `struct{}{}` other than to occupy no memory space. It could have been `map[string]bool`

Reply:
Yea, values are not needed here, we're just checking existence on keys. And a map is used so ordering of resource names isn't significant